File: sample/javascript/text-stream/command_r.js
import { CohereClient } from "cohere-ai";

const cohere = new CohereClient({
  token: process.env.CO_API_KEY,
});

(async () => {
  const chatStream = await cohere.chatStream({
    chatHistory: [
      { role: "USER", message: "Who discovered gravity?" },
      {
        role: "CHATBOT",
        message:
          "The man who is widely credited with discovering gravity is Sir Isaac Newton",
      },
    ],
    message: "What year was he born?",
  });

  for await (const message of chatStream) {
    if (message.eventType === "text-generation") {
      process.stdout.write(message);
    }
  }
})();

File: sample/python/text-stream/command_r.py
import cohere
import os

API_KEY = os.getenv("CO_API_KEY")

co = cohere.Client(API_KEY)

response = co.chat_stream(
    message="こんにちは、芝浦工業大学ShibaLabについて教えてください。",
    # ウェブ検索の結果を含めることができる
    connectors=[{"id": "web-search"}],
)

for event in response:
    if event.event_type == "text-generation":
        print(event.text, end="")
